Wikipedia/Code/SearchViewController.swift

```diff
@@ -532,12 +532,9 @@ class SearchViewController: ArticleCollectionViewController, UISearchBarDelegate
     override func articleURL(at indexPath: IndexPath) -> URL? {
         // This guard checks prevents an out of bounds exception, this is due to
```

Comment:
I could totally be reading this wrong, but I think we should revert this method and the method below back to what it originally was, `return nil`. It seems to me that `SearchViewController`'s `collectionView` only displays a list of the persisted recent search terms, whereas `SearchResultsViewController` only displays a list of search results from the network. We shouldn't be mixing and matching these data sources.

These two methods are here because it's superclass asserts false if they aren't implemented. Really cells on `SearchViewController` have little to do with articles, it's just persisted terms they have searched before. In hindsight we probably should never have subclassed `SearchViewController` from `ArticleCollectionViewController`, because we inherit and are expected to implement unnecessary methods like these.

Reply:
These have been reverted.